### PR TITLE
Refactor lib/private/Remote

### PR DESCRIPTION
**Pull request by shdehnavi**
## Summary
The required adjustments have been made to the classes in `/lib/private/Remote ` namespace.

The improvements:

- Using PHP8's constructor property promotion
- Adding return types
- Adding throws types
- Adding types to properties
- Using PHP8's `match` expression instead of `switch`

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits

### CODE REVIEW

File: lib/private/Remote/Instance.php [hunk 1]
```diff
@@ -32,55 +32,53 @@
...
 		return $this->getProtocol() . '://' . $this->getUrl();
 	}
 
 	/**
 	 * @return string The full version string in '13.1.2.3' format
+	 * @throws NotFoundException
 	 */
-	public function getVersion() {
+	public function getVersion(): string {
 		$status = $this->getStatus();
```
Comment (ChristophWurst) [inline lib/private/Remote/Instance.php:73]: getStatus needs finer type annotation for Psalm to acknowledge that this method will always return a string

Comment (ChristophWurst) [reply lib/private/Remote/Instance.php:73]: status might be `null`

Comment (shdehnavi) [reply lib/private/Remote/Instance.php:73]: It's addressed in the next commit

File: lib/private/Remote/User.php [hunk 2]
```diff
@@ -38,102 +38,100 @@ class User implements IUser {
...
-	public function getLanguage() {
+	public function getLanguage(): string {
 		return $this->data['language'];
 	}
 
 	/**
 	 * @return int
 	 */
-	public function getUsedSpace() {
+	public function getUsedSpace(): int {
```
Comment (nickvergessen) [inline lib/private/Remote/User.php:113]: this and all other quota related fields can also be float

Comment (shdehnavi) [reply lib/private/Remote/User.php]: It's addressed in the next commit

File: lib/private/Remote/Credentials.php [hunk 3]
```diff
@@ -25,31 +25,27 @@
...
 class Credentials implements ICredentials {
-	/** @var string */
-	private $user;
-	/** @var string */
-	private $password;
-
 	/**
 	 * @param string $user
 	 * @param string $password
 	 */
```
Comment (nickvergessen) [inline lib/private/Remote/Credentials.php:31]:
```suggestion
```

File: lib/private/Remote/Credentials.php [hunk 4]
```diff
@@ -25,31 +25,27 @@
...
-		$this->password = $password;
+	public function __construct(
+		private string $user,
+		private string $password,
+	) {
 	}
 
 	/**
 	 * @return string
 	 */
```
Comment (nickvergessen) [inline lib/private/Remote/Credentials.php:40]:
```suggestion
```

File: lib/private/Remote/Credentials.php [hunk 5]
```diff
@@ -25,31 +25,27 @@
...
-	public function getUsername() {
+	public function getUsername(): string {
 		return $this->user;
 	}
 
 	/**
 	 * @return string
 	 */
```
Comment (nickvergessen) [inline lib/private/Remote/Credentials.php:47]:
```suggestion
```

File: lib/private/Remote/Instance.php [hunk 6]
```diff
@@ -32,55 +32,53 @@
...
+	private ?array $status;
 
 	/**
 	 * @param string $url
 	 * @param ICache $cache
 	 * @param IClientService $clientService
 	 */
-	public function __construct($url, ICache $cache, IClientService $clientService) {
+	public function __construct(
+		$url,
```
Comment (nickvergessen) [inline lib/private/Remote/Instance.php:46]:
```suggestion
		string $url,
```

File: lib/private/Remote/User.php [hunk 7]
```diff
@@ -38,102 +38,100 @@ class User implements IUser {
...
-		$this->data = $data;
+	public function __construct(
+		private array $data,
+	) {
 	}
 
 
 	/**
 	 * @return string
 	 */
```
Comment (nickvergessen) [inline lib/private/Remote/User.php:49]:
```suggestion
```

Also on all other getters.

**Review by nickvergessen — CHANGES_REQUESTED** [review]

File: lib/private/Remote/User.php [hunk 8]
```diff
@@ -38,102 +38,97 @@ class User implements IUser {
...
-	 * @return string
-	 */
-	public function getUserId() {
+	public function getUserId(): string {
 		return $this->data['id'];
 	}
 
 	/**
 	 * @return string
 	 */
```
Comment (nickvergessen) [inline lib/private/Remote/User.php:53]:
```suggestion
```

Also the rest of the file

**Pull request closed without merging**